### ISSUE TITLE
Update CI and Docker images to Ubuntu 22.04

### ISSUE THREAD

**Issue by orangejulius** [pelias/pelias#951]
Pelias relies on Ubuntu for it's CI and Docker images, and we're currently still running Ubuntu 18.04. This version will have [LTS support through 2028](https://ubuntu.com/about/release-cycle), but Github Actions stopped using any Ubuntu 18.04 runners in [April 2023](https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/).

Most of our CI uses Ubuntu 20.04, but our Docker images are still built on Ubuntu 18.04, so we should ideally move everything to Ubuntu 22.04 (as suggested by @missinglink in https://github.com/pelias/docker-baseimage/pull/20) to stay current and ensure we don't have to mess with much for while longer.

**Comment by orangejulius** [comment]
@missinglink I'm probably going to spend a little time this weekend updating all our CI configs and Docker base image to use newer ubuntu versions. I agree with you we might as well go straight to Ubuntu 22 in general, and we should select Ubuntu 22 for our Docker baseimage.

Do you think there's any value in keeping Ubuntu 20 in our CI as well? It could be nice to support both but would mean more jobs in the matrix.

Also I'm thinking I'll use Github Actions [variables](https://docs.github.com/en/actions/learn-github-actions/variables) so we can store the Ubuntu version we use across all projects in _one_ place, and all our CI scripts can be updated without needing a commit to each repo. How's that sound?

**Comment by missinglink** [comment]
Sounds good, I haven't used variables but it sounds like a nice way of reducing the work to edit Configs across the various repos.

I don't see a huge value in keeping ubu 20 in the matrix, for docker we will likely settle on the latest stable OS quickly and for bare-metal, people run really old versions, one of my servers is still on an older version than that, so we'd never be able to cover all of them.

That said if we can do it and it's easy then why not, but if it requires a lot of conditional statements to support both i'd prefer to simple support the latest.

**Comment by orangejulius** [comment]
I've started testing out the GitHub Actions variable based approach to setting the Ubuntu version, it seems to work well and should be convenient going forward.

One annoying thing is we may have to reset a lot of our branch protection rules. If a `master` branch requires a particular unit test run to pass, it appears that the requirement pins to a specific nodejs _and_ Ubuntu version. So we may have to manually update all of these.
<img width="914" alt="Screen Shot 2023-07-14 at 5 53 51 PM" src="https://github.com/pelias/pelias/assets/111716/d872ae99-7617-4bf3-863f-a29c534a7acb">

**Comment by orangejulius** [comment]
Okay, I've gone through all our repos with Docker images and made sure they build on `ubuntu-22`. There was only [one](https://github.com/pelias/libpostal-service/pull/8) minor change required.

All our CI for our primary repos (services, importers, schema, etc) is now running ubuntu 22 as well. All the supporting library repos (like pelias/query, dbclient, etc) have yet to be updated but those are lower priority since they are implicitly tested by the primary repos. So we can get to them someday :)

Using the script below, run from a directory with all the pelias repos checked out, I can confirm everything builds on Ubuntu 22, so https://github.com/pelias/docker-baseimage/pull/27 is now good to merge. After that merges, I'll kick off new Docker image builds using a no-op commit first to the [docker-libpostal_baseimage](https://github.com/pelias/docker-libpostal_baseimage/) repo (since its Docker image is a dependency for others), and then to all the other repos.

Then we should be all 100% moved to Ubuntu 22! 🥳 

<details>
<summary>Script to check that all images build</summary>

```sh
#!/bin/bash

set -euo pipefail

importers="openstreetmap whosonfirst openaddresses polylines geonames transit csv-importer"
services="api placeholder libpostal-service interpolation pip-service"
libpostal_baseimage="docker-libpostal_baseimage"
schema="schema"

modules="pbf2json model query dbclient"
repos="$importers $services $schema"

echo $repos

for repo in $repos; do
        echo $repo
        pushd $repo
        pwd
        git stash
        git checkout master
        git fetch
        git pull
        git cleanup || true

        branch_name="docker-ubuntu-22"
        git branch -D $branch_name || true
        git checkout -b $branch_name || true

        sed -i 's/pelias\/baseimage/pelias\/baseimage\:ubuntu-22/' Dockerfile
        sed -i 's/pelias\/libpostal_baseimage/pelias\/libpostal_baseimage\:ubuntu-22/' Dockerfile

        docker build . -t pelias/$repo:ubuntu-22

        popd
done
```
</details>

**Comment by orangejulius** [comment]
So, I never really finished this. I did an audit today and CI is running on Ubuntu 16, 18, 20 and 22 across different Pelias repos 😨.

@missinglink also pointed out in https://github.com/pelias/whosonfirst/pull/538#issuecomment-1711519077 that the `UBUNTU_VERSION` variable breaks forks. It didn't buy us much anyway, so I'm going to open some PRs combined with other CI and ubuntu / node.js version cleanup to fix it.

**Comment by orangejulius** [comment]
Ok, I've gone through every one of our repos, they are all using ubuntu-22. As mentioned in https://github.com/pelias/pelias/issues/951#issuecomment-1636482672 I did have to modify a few branch protection rules. We always have "status checks must pass", which is what we want. However, for a few repos we also had _specific_ checks required to pass, which often specified an EOL Ubuntu or Node.js version. I've removed all those and now everything is good.
